### PR TITLE
Edit, Index and Destroy post from user

### PR DESCRIPTION
**Pull request by MikeHoang**

### CODE REVIEW

File: app/controllers/user/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  def index
+    @posts = current_user.posts.all
+  end
+
+  def show
+    @post = current_user.posts.find_by(id: params[:id])
+  end
+
+  def new
+    @post = current_user.posts.build(post_params)
```
Comment (nguyenductoan) [inline app/controllers/user/posts_controller.rb]: you can use `@posts =  current_user.posts.new ` to new instance.

File: app/controllers/user/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
+class User::PostsController < ApplicationController
+
+
+  def index
+    @posts = current_user.posts.all
+  end
+
+  def show
+    @post = current_user.posts.find_by(id: params[:id])
```
Comment (nguyenductoan) [inline app/controllers/user/posts_controller.rb]: change `current_user.posts.find_by(id: params[:id])` to `current_user.posts.find(params[:id])`

File: app/controllers/user/posts_controller.rb (hunk 1 above)
Comment (lhhieu89) [inline app/controllers/user/posts_controller.rb]: @nguyenductoan : Why we don't use
```ruby
@post = Post.new
```

File: app/views/user/posts/index.html.slim [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          th Update At
+          th
+      tbody
+
+        - @posts.each do |post|
+          tr
+            td = post.title
+            td = post.created_at
+            td = post.updated_at
+
```
Comment (nguyenductoan) [inline app/views/user/posts/index.html.slim]: Please remove empty lines!

File: app/models/partner.rb [hunk 4]
```diff
@@ -37,4 +37,5 @@
 
 class Partner < User
   mount_uploader :avatar, ImageUploader
+  has_many :student
```
Comment (lhhieu89) [inline app/models/partner.rb]: It shoud be `has_many :students`

File: .byebug_history [hunk 5]
```diff
@@ -1,3 +1,15 @@
...
+post_params.class
+post_params.last
+post_params.count
+post_params.image
+post_params
+@post.image
+@post
 quit
 @student.save!
 @student.save
```
Comment (lhhieu89) [inline .byebug_history]: Please remove this file

File: Gemfile [hunk 6]
```diff
@@ -38,6 +38,7 @@ gem 'mini_magick'
 #upload images
 gem 'carrierwave'
 gem 'truncate_html'
+gem 'fog',                     '1.36.0'
```
Comment (lhhieu89) [inline Gemfile]: Why did you need this gem? i didn't see you use it

File: app/controllers/admin/partners_controller.rb [hunk 7]
```diff
@@ -1,12 +1,16 @@
 class Admin::PartnersController < Admin::AdminController
-
+  before_action :authenticate_user!
   before_action :admin_user
```
Comment (lhhieu89) [inline app/controllers/admin/partners_controller.rb]: i thinks we need only one function from two functions above

File: app/controllers/admin/partners_controller.rb [hunk 8]
```diff
@@ -29,7 +33,7 @@ def destroy
   private
 
   def get_partner
-    @partner = Partner.find(params[:id])
+    @partner = Partner.find_by(id: params[:id])
```
Comment (lhhieu89) [inline app/controllers/admin/partners_controller.rb]: You should to use `find` because it will raise the error if NOT FOUND, `find_by` will return `nil`

File: app/controllers/user/posts_controller.rb [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    @post = current_user.posts.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      flash[:success] = "post created!"
+      redirect_to user_posts_path
+    else
+      redirect_to root_url
```
Comment (lhhieu89) [inline app/controllers/user/posts_controller.rb]: You should to render new page again if @post is invalid

File: app/controllers/user/posts_controller.rb [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    if @post.update_attributes(post_params)
+      flash[:success]= "Update successfully"
+      redirect_to user_posts_path
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    Post.find(params[:id]).destroy
```
Comment (lhhieu89) [inline app/controllers/user/posts_controller.rb]: Please create the new function to get current post and call it at `before_action`
Refer: `app/controllers/admin/banners_controller.rb`

File: app/controllers/user/posts_controller.rb [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    flash[:success] = "Post is deleted"
+    redirect_to user_posts_path
+  end
+
+  private
+   def post_params
+      params.require(:post).permit(:title, :brief,
+                                   :image, :remove_image, :content, :public)
+    end
```
Comment (lhhieu89) [inline app/controllers/user/posts_controller.rb]: Please reindent your code

File: app/models/post.rb [hunk 12]
```diff
@@ -18,6 +18,14 @@ class Post < ActiveRecord::Base
...
+
+	def image_size
+			if image.size > 5.megabytes
+        errors.add(:image, "should be less than 5MB")
+      end
+
+	end
 
 	def self.top_reading(num = 6)
 		Post.limit(num)
```
Comment (lhhieu89) [inline app/models/post.rb]: Fix indentation in this file, too

File: db/seeds.rb [hunk 13]
```diff
@@ -28,7 +28,7 @@
...
...=\"575\" src=\"http://www.studentlifecare.com/wp-content/uploads/The-Mint-Courtyard-Auditorium-Bar.-Photo-c-Penelope-Beveridge-1024x681.jpg\" width=\"802\" /></a></p>\r\n\r\n<p>&nbsp;</p>\r\n\r\n<p>B&ecirc;n cạnh những yếu tố tuyệt vời th&igrave; song h&agrave;nh với n&oacute; l&agrave; điều kiện sống v&agrave; mức ph&iacute; sinh hoạt kh&aacute; cao. Đ&acirc;y l&agrave; 1 trong số những trở ngại lớn cho sinh vi&ecirc;n học tập tại đ&acirc;y</p>\r\n\r\n<p>Mức sống c&ugrave;ng chi ph&iacute; học tập cao đ&ograve;i hỏi c&aacute;c bạn sinh vi&ecirc;n cần đi l&agrave;m th&ecirc;m để duy tr&igrave; cuộc sống. Danielle Smith &ndash; 22 tuổi, du học sinh tại trường University of Sydney vẫn đi l&agrave;m th&ecirc;m mặc d&ugrave; bạn đang bước v&agrave;o k&igrave; học cuối tại trường.</p>\r\n\r\n<p>Tuy vậy vẫn c&ograve;n nhiều c&aacute;ch gi&uacute;p bạn giải quyết b&agrave;i to&aacute;n kinh ph&iacute; khi cơ hội việc l&agrave;m đa dạng, share ph&ograve;ng trọ hay nấu nướng tập thể. Đ&acirc;y cũng l&agrave; c&aacute;ch nhiều sinh vi&ecirc;n lựa chọn để san sẻ bớt kinh ph&iacute;, d&agrave;nh thời gian cho việc học tập v&agrave; thực tập.</p>\r\n", public: true, category_id: 2, user_id: 1},
```
Comment (lhhieu89) [inline db/seeds.rb]: Please remove all file in `public/uploads/*` and don't push it to git

File: app/controllers/user/posts_controller.rb (hunk 10 above)
Comment (lhhieu89) [inline app/controllers/user/posts_controller.rb]: Use `@post.destroy` instead
